src/biogeophys/InfiltrationExcessRunoffMod.F90

```diff
@@ -0,0 +1,342 @@
+
+  !-----------------------------------------------------------------------
+  subroutine ComputeQinmaxVic(bounds, num_hydrologyc, filter_hydrologyc, &
+       soilhydrology_inst, &
+       fsat, qflx_in_soil, qinmax_on_unsaturated_area)
+    !
+    ! !DESCRIPTION:
+    ! Compute qinmax using the VIC parameterization
+    !
+    ! Citation: Wood et al. 1992, "A land-surface hydrology parameterization with subgrid
+    ! variability for general circulation models", JGR 97(D3), 2717-2728.
+    !
+    ! !ARGUMENTS:
+    type(bounds_type), intent(in) :: bounds
+    integer, intent(in) :: num_hydrologyc       ! number of column soil points in column filter
+    integer, intent(in) :: filter_hydrologyc(:) ! column filter for soil points
+    type(soilhydrology_type) , intent(in) :: soilhydrology_inst
+    real(r8) , intent(in)    :: fsat( bounds%begc: )                       ! fractional area with water table at surface
+    real(r8) , intent(in)    :: qflx_in_soil( bounds%begc: )               ! surface input to soil (mm/s)
+    real(r8) , intent(inout) :: qinmax_on_unsaturated_area( bounds%begc: ) ! maximum infiltration capacity on the unsaturated fraction of the column (mm H2O /s)
+    !
+    ! !LOCAL VARIABLES:
+    integer  :: fc, c
+    real(r8) :: dtime       ! land model time step (sec)
+    real(r8) :: top_icefrac ! ice fraction in top VIC layers
+    real(r8) :: max_infil   ! max infiltration capacity in VIC (mm)
+    real(r8) :: i_0         ! average soil moisture in top VIC layers (mm)
+    real(r8) :: rsurf_vic   ! VIC surface runoff
+    real(r8) :: basis       ! variable soil moisture holding capacity in top VIC layers for runoff calculation
+
+    character(len=*), parameter :: subname = 'ComputeQinmaxVic'
+    !-----------------------------------------------------------------------
+
+    SHR_ASSERT_ALL((ubound(fsat) == (/bounds%endc/)), errMsg(sourcefile, __LINE__))
+    SHR_ASSERT_ALL((ubound(qflx_in_soil) == (/bounds%endc/)), errMsg(sourcefile, __LINE__))
+    SHR_ASSERT_ALL((ubound(qinmax_on_unsaturated_area) == (/bounds%endc/)), errMsg(sourcefile, __LINE__))
+
+    associate( &
+          top_max_moist    =>    soilhydrology_inst%top_max_moist_col, & ! Input:  [real(r8) (:)   ]  maximum soil moisture in top VIC layers
+          top_moist        =>    soilhydrology_inst%top_moist_col    , & ! Input:  [real(r8) (:)   ]  soil moisture in top VIC layers
+          top_ice          =>    soilhydrology_inst%top_ice_col      , & ! Input:  [real(r8) (:)   ]  ice len in top VIC layers
+          b_infil          =>    soilhydrology_inst%b_infil_col        & ! Input:  [real(r8) (:)   ]  VIC b infiltration parameter
+          )
+
+    dtime = get_step_size()
+
+    do fc = 1, num_hydrologyc
+       c = filter_hydrologyc(fc)
+       top_icefrac = min(1._r8,top_ice(c)/top_max_moist(c))
+       max_infil = (1._r8+b_infil(c)) * top_max_moist(c)
+       i_0 = max_infil * (1._r8 - (1._r8 - fsat(c))**(1._r8/b_infil(c)))
```

Comment:
**Comment from @martynpclark Aug 24, 2017**

Note that all of this code is the analytical solution to compute change in storage associated with surface runoff. This code should be in the saturation-excess module.